c10/cuda/CUDAStream.h

```diff
@@ -213,6 +213,17 @@ CAFFE2_API CUDAStream getCurrentCUDAStream(DeviceIndex device_index = -1);
  */
 CAFFE2_API void setCurrentCUDAStream(CUDAStream stream);
 
+/**
+ * Register an external CUDA stream. This will return a CUDAStream that
+ * points to the stream registered. Rationale behind this API is that in the
+ * current scheme, CUDAStream does not contain the actual stream. Hence
+ * preventing users from just wrapping their own cudaStream_t into a CUDAStream
+ * type. This API is compatiable with 'CUDAStreamGuard'.
+ * 
+ * Caller is responsible for the lifetime of the stream. This may change in the future.
```

Comment:
If you're OK with the limitations I described above, the API docs here need to more clearly describe the caveats.

Reply:
I'll update the doc